core/chains/evm/txm/storage/inmemory_store.go

```diff
@@ -0,0 +1,333 @@
+package storage
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/txm/types"
+)
+
+const (
+	maxQueuedTransactions = 250
+	pruneSubset           = 3
+)
+
+type InMemoryStore struct {
+	sync.RWMutex
+	lggr      logger.Logger
+	txIDCount uint64
+	address   common.Address
+	chainID   *big.Int
+
+	UnstartedTransactions   []*types.Transaction
+	UnconfirmedTransactions map[uint64]*types.Transaction
+	ConfirmedTransactions   map[uint64]*types.Transaction
+	FatalTransactions       []*types.Transaction
+
+	Transactions map[uint64]*types.Transaction
+}
+
+func NewInMemoryStore(lggr logger.Logger, address common.Address, chainID *big.Int) *InMemoryStore {
+	return &InMemoryStore{
+		lggr:                    logger.Named(lggr, "InMemoryStore"),
+		address:                 address,
+		chainID:                 chainID,
+		UnconfirmedTransactions: make(map[uint64]*types.Transaction),
+		ConfirmedTransactions:   make(map[uint64]*types.Transaction),
+		Transactions:            make(map[uint64]*types.Transaction),
+	}
```

Comment:
Couldn't we also initialize `UnstartedTransactions` here to avoid reallocations?
```suggestion
	return &InMemoryStore{
		lggr:                    logger.Named(lggr, "InMemoryStore"),
		address:                 address,
		chainID:                 chainID,
		UnstartedTransactions:   make([]*types.Transaction, 0, maxQueuedTransactions),
		UnconfirmedTransactions: make(map[uint64]*types.Transaction),
		ConfirmedTransactions:   make(map[uint64]*types.Transaction),
		Transactions:            make(map[uint64]*types.Transaction),
	}
```

Reply:
That's a valid comment, although it comes down to what is the expected average size of the queue at runtime. For example, it's not worth pre-allocating max size if the queue only stores 5-10 transactions at a time. I can perhaps run a benchmark test to measure this.